singularity-eos/eos/eos_spiner_sie_transforms.hpp

```diff
@@ -0,0 +1,214 @@
+
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto transform(const Real e, Args &&...) const {
+    return e;
+  }
+
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto inverse(const Real e_transformed, Args &&...) const {
+    return e_transformed;
+  }
+};
+
+template <typename Data>
+struct ShiftTransform {
+  PORTABLE_INLINE_FUNCTION  ShiftTransform() = default;
+  
+  template <typename DataT_in>
+  PORTABLE_INLINE_FUNCTION ShiftTransform(const DataT_in &data) : data_(data) {}
+
+ private:
+  Data data_;
+
+ public:
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto transform(const Real e, const Real rho,
+                                          Args &&...) const {
+    const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+    const Real e_cold = data_.sieCold.interpToReal(lRho);
+    return e - e_cold;
+  }
+
+  template <typename... Args>
+  PORTABLE_INLINE_FUNCTION auto inverse(const Real e_transformed, const Real rho,
+                                        Args &&...) const {
+    const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);   
+    const Real e_cold = data_.sieCold.interpToReal(lRho);
+    return e_transformed + e_cold;
+  }
+};
+
+
+
+//Divide by the heat capacity
+template<typename Data>
+struct DivideByCvTransform {
+  PORTABLE_INLINE_FUNCTION  DivideByCvTransform() = default;
+  
+  template <typename DataT_in>  
+  PORTABLE_INLINE_FUNCTION DivideByCvTransform(const DataT_in &data) : data_(data) {}
+
+  private: 
+    Data data_;
+    Real min_Cv_ = 1.0e-08;
+
+  public:
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+    auto transform(Real e, Real rho, Args &&...) const {
+        const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+	const Real lE   = spiner_common::to_log(e, data_.lEOffset);
+        Real Cv = 1./data_.dTdE.interpToReal(lRho, lE);
+        Cv = std::max(Cv, min_Cv_);
+        return e / Cv;
+    }
+
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+    auto inverse(Real e_transformed, Real rho, Real e_orig, Args &&...) const {
+        const Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+	const Real lE   = spiner_common::to_log(e_orig, data_.lEOffset);
+	Real Cv = 1./data_.dTdE.interpToReal(lRho, lE);
+        Cv = std::max(Cv, min_Cv_);
+        return e_transformed * Cv;
+    }
+};
+
+
+
+//Divide by T^alpha
+template<typename Data>
+struct ScaleTransform {
+    PORTABLE_INLINE_FUNCTION  ScaleTransform() = default;
+    
+    template <typename DataT_in>
+PORTABLE_INLINE_FUNCTION ScaleTransform(const DataT_in &data) : data_(data) {}
+
+private:
+    Data data_;
+    Real min_T_ = 1.0e-08; //Good minimum temperature? 
+
+public: 
+    template <typename... Args>
+    PORTABLE_INLINE_FUNCTION
+    auto transform(Real e, Real rho, Args &&...) const {
+        Real lRho = spiner_common::to_log(rho, data_.lRhoOffset);
+        Real T = data_.T.interpToReal(lRho); //T?
```

Comment:
This one isn't going to work, because T depends on rho and sie. Let's add a TODO to this one and table it for now. I think it will require auxiliary databoxes to be available.